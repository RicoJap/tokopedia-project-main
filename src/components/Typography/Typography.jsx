import React from "react";
import MuiTypography from "@material-ui/core/Typography";


const Typography = ({label, variant, ...attributes}) => {
    return <MuiTypography variant={variant} {...attributes}>{label}</MuiTypography>
}

export default Typography;