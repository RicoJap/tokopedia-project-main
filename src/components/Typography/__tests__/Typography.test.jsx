import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import Typography from "../Typography";

const MockTypography = () => {
  return <Typography variant="p" label="Test Label" />;
};

describe("Snapshot testing", () => {
  const tree = renderer.create(<MockTypography />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Typography Component renders correctly", () => {
  test("Check if Typography component is present in the document", () => {
    render(<MockTypography />);
    expect(screen.getByText("Test Label")).toBeInTheDocument();
  });
});
