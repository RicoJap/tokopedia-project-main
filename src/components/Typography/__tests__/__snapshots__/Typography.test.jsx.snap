// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<span
  className="MuiTypography-root"
>
  Test Label
</span>
`;
